cinema/views.py

```diff
@@ -1,10 +1,21 @@
 from datetime import datetime
 
 from django.db.models import F, Count
-from rest_framework import viewsets
+from rest_framework import viewsets, mixins
+from rest_framework.authentication import TokenAuthentication
```

Comment:
The use of `TokenAuthentication` is consistent across most viewsets, but ensure this aligns with your application's authentication strategy. If different authentication methods are needed, adjust accordingly.